gwctl/pkg/policymanager/manager.go

```diff
@@ -230,29 +229,53 @@ func (p Policy) ClientObject() client.Object { return p.Unstructured() }
 func PolicyFromUnstructured(u unstructured.Unstructured, policyCRDs map[PolicyCrdID]PolicyCRD) (Policy, error) {
 	result := Policy{u: u}
 
-	// Identify targetRef of Policy.
+	// Identify targetRefs of Policy.
 	type genericPolicy struct {
 		metav1.TypeMeta   `json:",inline"`
 		metav1.ObjectMeta `json:"metadata,omitempty"`
 		Spec              struct {
-			TargetRef gatewayv1alpha2.NamespacedPolicyTargetReference
+			TargetRefs []gatewayv1alpha2.NamespacedPolicyTargetReference `json:"targetRefs"`
+			TargetRef  *gatewayv1alpha2.NamespacedPolicyTargetReference  `json:"targetRef"`
 		}
 	}
 	structuredPolicy := &genericPolicy{}
 	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(u.UnstructuredContent(), structuredPolicy); err != nil {
 		return Policy{}, fmt.Errorf("failed to convert unstructured policy resource to structured: %v", err)
 	}
-	result.targetRef = common.ObjRef{
-		Group:     string(structuredPolicy.Spec.TargetRef.Group),
-		Kind:      string(structuredPolicy.Spec.TargetRef.Kind),
-		Name:      string(structuredPolicy.Spec.TargetRef.Name),
-		Namespace: structuredPolicy.GetNamespace(),
-	}
-	if result.targetRef.Namespace == "" {
-		result.targetRef.Namespace = result.u.GetNamespace()
+
+	// Process targetRefs
+	for _, targetRef := range structuredPolicy.Spec.TargetRefs {
+		objRef := common.ObjRef{
+			Group:     string(targetRef.Group),
+			Kind:      string(targetRef.Kind),
+			Name:      string(targetRef.Name),
+			Namespace: u.GetNamespace(),
+		}
+		if objRef.Namespace == "" {
+			objRef.Namespace = u.GetNamespace()
+		}
```

Comment:
I may be missing something, but I understand `objRef.Namespace` has been initialized to `u.GetNamespace()` already, no? Then, if it's an empty string, it's set again to the same `u.GetNamespace()`? Maybe you meant:

```suggestion
			Namespace: structuredPolicy.GetNamespace(),
		}
		if objRef.Namespace == "" {
			objRef.Namespace = u.GetNamespace()
		}
```